src/Result.php

```diff
@@ -0,0 +1,183 @@
+<?php
+declare(strict_types=1);
+
+namespace FastRoute;
+
+use ArrayAccess;
+use RuntimeException;
+
+/**
+ * Result Object
+ *
+ * @implements ArrayAccess<int, mixed>
+ */
+class Result implements ArrayAccess
```

Comment:
Do we really need to make this behave as an array? What's the added value?

Reply:
Also, what's the added value of enabling inheritance here? To me this is basically a transport object and could be `final`.

Reply:
It's a lib and people might alter it and want to change the object. I'm not a big fan of using final classes and private properties in libs but prefer to keep them open. My experience is that even if people think "nobody" ever needs it, there are cases in which people will change it for a good reason.

Reply:
This is essentially a DTO. It can be final. If we want people to extend this we should provide an interface. I'd prefer to start the process with just the object. 

This patch breaks BC by renaming the classes, it doesn't make sense not to simplify things with that argument.